internal/activities/validate_structure.go

```diff
@@ -171,3 +176,29 @@ func extraNodes(sipBase, path string, expected []string, matchDir bool) ([]strin
 
 	return extras, nil
 }
+
+func validateName(name string) bool {
+	validChars := "-_.()"
+
+	// Generate string containing valid characters.
+	validChars = appendRuneRange(validChars, 'a', 'z')
+	validChars = appendRuneRange(validChars, 'A', 'Z')
+	validChars = appendRuneRange(validChars, '0', '9')
```

Comment:
This is nice, but I think it would be a lot clearer with a long string constant. It will also prevent from calculating it each time.